pkg/manager/impl/validation/task_validator.go

```diff
@@ -43,6 +42,26 @@ func validateContainer(task core.TaskTemplate, taskConfig runtime.TaskResourceCo
 	return nil
 }
 
+func validatedK8sPodSpec(task *core.TaskTemplate) error {
+	if task.GetK8SPod().PodSpec == nil {
+		return errors.NewFlyteAdminErrorf(codes.InvalidArgument, "K8sPod task type targets must specify a non-empty pod spec")
+	}
+	var podSpec v1.PodSpec
+	jsonObj, err := json.Marshal(task.GetK8SPod().PodSpec)
+	if err != nil {
+		return errors.NewFlyteAdminErrorf(codes.InvalidArgument, "Failed to json marshal K8sPod task type target with err: %v", err)
+	}
+	if err = json.Unmarshal(jsonObj, &podSpec); err != nil {
+		return errors.NewFlyteAdminErrorf(codes.InvalidArgument, "Failed to json unmarshal K8sPod task type target with err: %v", err)
+	}
```

Comment:
Should we move this pattern somewhere? json marshal json unmarshal... maybe here: https://github.com/flyteorg/flytestdlib/blob/master/utils/marshal_utils.go but it's ok if it's within flyteadmin utils somewhere?

Reply:
@EngHabu done https://github.com/flyteorg/flytestdlib/pull/90, please review